src/lib/NonlinearDutchDecayLib.sol

```diff
@@ -46,7 +45,35 @@ library NonlinearDutchDecayLib {
         (uint16 startPoint, uint16 endPoint, int256 relStartAmount, int256 relEndAmount) =
             locateCurvePosition(curve, blockDelta);
         // get decay of only the relative amounts
-        int256 curveDelta = DutchDecayLib.linearDecay(startPoint, endPoint, blockDelta, relStartAmount, relEndAmount);
+        int256 curveDelta = v3LinearInputDecay(startPoint, endPoint, blockDelta, relStartAmount, relEndAmount);
+
+        return startAmount.boundedSub(curveDelta, minAmount, maxAmount);
+    }
+
+    function decayOutput(
```

Comment:
Can we combine these two functions to reduce code duplication?

Reply:
We can pass in a bool to differentiate input vs output

Reply:
This was the approach I took in my first commit, but took @zhongeric's [suggestion](https://github.com/Uniswap/UniswapX/pull/305#discussion_r1786490670) to not use a boolean and separate out instead